tests/scripts/parse_notice.py

```diff
@@ -0,0 +1,277 @@
+
+        self.read_requirements()
+
+        scanned_results_data = self.read_content_from_file(self.scanned_json_file)
+
+        if not scanned_results_data:
+            raise ValueError(f"{self.scanned_json_file} is empty")
+
+        self.scanned_results_json: Any = json.loads(scanned_results_data)
+
+        notice_file_content: str = self.read_content_from_file("NOTICE.txt")
+
+        package_pattern = r"(?:Package: ([^\n]+))"
+        existing_packages: list[str] = re.findall(package_pattern, notice_file_content)
+        existing_packages.sort()
+
+        requirements_name_from_file = [requirement for requirement in self.required_packages.keys()]
+        requirements_name_from_file.sort()
+
+        real_requirements_name = [requirement for requirement in self.required_packages.values()]
+        real_requirements_name.sort()
+
+        if self.mode == "--check":
+            if real_requirements_name == existing_packages:
+                print("[!] There is no new package listed in the requirements files")
+                sys.exit()
+            else:
+                for new_package in requirements_name_from_file:
+                    if self.required_packages[new_package] not in existing_packages:
+                        real_package_name: str = self.required_packages[new_package]
+                        print(f"[!] New package found: '{real_package_name}'")
+
+                print("[!] Run the script with the 'fix' argument in order to add it to the NOTICE.txt file")
+                sys.exit(1)
+
+        elif self.mode == "--fix":
+            if real_requirements_name == existing_packages:
+                print("[!] There is no new package listed in the requirements files")
+                sys.exit()
+
+            for new_package in requirements_name_from_file:
+                if self.required_packages[new_package] not in existing_packages:
+                    real_package_name = self.required_packages[new_package]
+
+                    print(f"[!] New package found: '{real_package_name}'")
+                    self.process_package(required_package=new_package)
+                    self.verify_license_in_packages(processed_package=new_package)
+
+                    processed_package = self.processed_packages.get(new_package)
+
+                    if not processed_package:
+                        print(
+                            f"[-] Nothing has been found for package '{real_package_name}' in {self.scanned_json_file}",
+                        )
+                        continue
+
+                    if (
+                        "package_name" not in processed_package
+                        or "version" not in processed_package
+                        or "homepage_url" not in processed_package
+                        or "license_name" not in processed_package
+                        or "license_path" not in processed_package
+                        or "license_content" not in processed_package
```

Comment:
for `license_content` we could add a lookup on https://spdx.org/licenses/ as extra step
 
not sure if, once we identify the license name, missing the content we should skip the entry at all.

Reply:
same as above. 

What would happen if we don't have anything for the license content for that package? Shall we add a general license based on the license name?